File: code_challenge_solution/python/106 Linked list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
    
class LinkedList:
    def __init__(self):
        self.head = None
    
    def append(self, data):
        new_node = Node(data)
        
        if not self.head:
            self.head = new_node
        else:
            curr_node = self.head
            while curr_node.next:
                curr_node = curr_node.next
            curr_node.next = new_node
    
    def prepend(self, data):
        new_node = Node(data)
        
        new_node.next = self.head
        self.head = new_node
    
    def delete(self, data):
        if not self.head:
            return
        
        if self.head.data == data:
            self.head = self.head.next
            return
        
        curr_node = self.head
        next_node = curr_node.next
        
        while next_node:
            if next_node.data == data:
                curr_node.next = next_node.next
                return
            curr_node = next_node
            next_node = next_node.next
    
    def print_list(self):
        if not self.head:
            print("Linked list is empty")
            return
        
        curr_node = self.head
        while curr_node:
            print(curr_node.data, end=" ")
            curr_node = curr_node.next
        
        print()

File: code_challenge_solution/c++/106 Linked list.cpp
```cpp
#include <iostream>

using namespace std;

// Node Class
class Node {
public:
    int data;
    Node* next;
};

// Linked List Class
class LinkedList {
private:
    Node* head;

public:
    LinkedList() {
        head = NULL;
    }

    // Insert a node at the beginning of the linked list
    void insertAtBeginning(int value) {
        Node* newNode = new Node();
        newNode->data = value;
        newNode->next = head;
        head = newNode;
    }

    // Insert a node at the end of the linked list
    void insertAtEnd(int value) {
        Node* newNode = new Node();
        newNode->data = value;
        newNode->next = NULL;

        // If list is empty, make the new node as head
        if (head == NULL) {
            head = newNode;
            return;
        }

        // Else traverse till the last node and insert the new node there
        Node* last = head;
        while (last->next != NULL) {
            last = last->next;
        }
        last->next = newNode;
    }

    // Print the linked list
    void printList() {
        Node* temp = head;
        while (temp != NULL) {
            cout << temp->data << " ";
            temp = temp->next;
        }
        cout << endl;
    }
};

int main() {
    LinkedList list;

    // Inserting nodes at the beginning
    list.insertAtBeginning(5);
    list.insertAtBeginning(3);
    list.insertAtBeginning(1);

    // Inserting nodes at the end
    list.insertAtEnd(7);
    list.insertAtEnd(9);

    // Printing the linked list
    list.printList();

    return 0;
}
```